pandas/core/internals/blocks.py

```diff
@@ -1383,6 +1383,9 @@ def interpolate(
         limit_area: Literal["inside", "outside"] | None = None,
         **kwargs,
     ) -> list[Block]:
+        valid = missing.NP_METHODS + missing.SP_METHODS
```

Comment:
A better place for this check (and the one you added in `get_interp_index`) in is probably in `_interpolate_scipy_wrapper`.

We want to make these checks at the lowest level possible

Reply:
thanks, I moved this check from `pandas/core/internals/blocks.py` to `_interpolate_scipy_wrapper` in `pandas/core/missing.py`.

I left the check in `get_interp_index`, otherwise, several tests fail.